physics/ephemeris_body.hpp

```diff
@@ -948,11 +956,21 @@ void Ephemeris<Frame>::AppendMassiveBodiesState(
     typename NewtonianMotionEquation::SystemState const& state) {
   last_state_ = state;
   int index = 0;
-  for (auto& trajectory : trajectories_) {
-    trajectory->Append(
+  for (int i = 0; i < trajectories_.size(); ++i) {
+    auto const& trajectory = trajectories_[i];
+    auto const status = trajectory->Append(
         state.time.value,
         DegreesOfFreedom<Frame>(state.positions[index].value,
                                 state.velocities[index].value));
+
+    // Handle the apocalypse.
+    if (!status.ok()) {
+      last_severe_integration_status_ =
+          Status(status.error(),
+                 "Error extending trajectory for " + bodies_[i]->name() + ". " +
+                     status.message());
```

Comment:
I'm not sure how apocalypses behave in practice; would several bodies undergo one during the same call to prolong? If so, it would help to list them all (that would point to a collision between them, rather than a collision between a body and *whatever*).